### PR TITLE
[ci skip] Add upgrade guide and update documentation

### PR DESCRIPTION
**Pull request by thefirstofthe300**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -157,6 +157,8 @@ data "google_container_engine_versions" "region" {
 }
 
 data "google_container_engine_versions" "zone" {
+  provider = "google-beta"
```
Comment (aaron-lane) [inline main.tf]: This change is not relevant to the branch topic.

File: modules/private-cluster/main.tf [hunk 2]
```diff
@@ -157,6 +157,8 @@ data "google_container_engine_versions" "region" {
 }
 
 data "google_container_engine_versions" "zone" {
+  provider = "google-beta"
```
Comment (aaron-lane) [inline modules/private-cluster/main.tf]: This change is not relevant to the branch topic.

File: modules/private-cluster/variables.tf [hunk 3]
```diff
@@ -126,7 +126,7 @@ variable "remove_default_node_pool" {
 
 variable "disable_legacy_metadata_endpoints" {
   description = "Disable the /0.1/ and /v1beta1/ metadata server endpoints on the node. Changing this value will cause all node pools to be recreated."
-  default     = true
+  default     = "true"
```
Comment (aaron-lane) [inline modules/private-cluster/variables.tf]: This change is not relevant to the branch topic.

File: variables.tf [hunk 4]
```diff
@@ -126,7 +126,7 @@ variable "remove_default_node_pool" {
 
 variable "disable_legacy_metadata_endpoints" {
   description = "Disable the /0.1/ and /v1beta1/ metadata server endpoints on the node. Changing this value will cause all node pools to be recreated."
-  default     = true
+  default     = "true"
```
Comment (aaron-lane) [inline variables.tf]: This change is not relevant to the branch topic.

File: main.tf (hunk 1 above)
Comment (thefirstofthe300) [reply main.tf]: This appears to be the result of a code skew between the autogen code and the post autogen'd code introduced by https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/commit/1d73e9d56ee0026aa09ddc4f8e6b2a2c325c1c93

tl;dr: I screwed up in a commit outside of this branch so regenerating the documentation makes a diff in the generated code unavoidable.

That being said, the diff should be for removing the provider definition from the `autogen/main.tf` file, not adding it here.

This should now be fixed.

File: modules/private-cluster/main.tf (hunk 2 above)
Comment (thefirstofthe300) [reply modules/private-cluster/main.tf]: Same as above.

File: modules/private-cluster/variables.tf (hunk 3 above)
Comment (thefirstofthe300) [reply modules/private-cluster/variables.tf]: Same as above.

File: variables.tf (hunk 4 above)
Comment (thefirstofthe300) [reply variables.tf]: This is also the result of a skew between the templates and autogen'd code in previous PR: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/commit/6813253b5f314a4282b231ce1489f0d65180e1d9

The diff is unavoidable when generating the documentation.
